foxpuppet/windows.py

```diff
@@ -2,44 +2,75 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
-        self._handle = None
 
     @property
-    def all(self):
-        """Returns all current window handles"""
-        return self.selenium.window_handles
+    def windows(self):
+        return [BaseWindow(self.selenium, handle=handle)
+                for handle in self.selenium.window_handles]
 
-    @property
-    def current(self):
-        """Returns current window handle"""
-        return self.selenium.current_window_handle
 
-    def focus(self, handle):
-        """Switch Selenium focus to a specific window"""
-        return self.selenium.switch_to.window(handle)
+class BaseWindow(object):
 
-    @contextmanager
-    def wait_for_new_window(self):
-        handles_before = self.selenium.window_handles
-        yield
-        WebDriverWait(self.selenium, 10).until(
-            lambda driver: len(handles_before) != len(driver.window_handles))
+    def __init__(self, selenium, handle, *args, **kwargs):
+        self.selenium = selenium
+        self._handle = handle
+
+    @property
+    def handle(self):
+        return self._handle
 
     @property
     def window_element(self):
-        """Returns the inner DOM window element.
+        """ Returns the inner DOM window element.
 
         :returns: DOM window element.
         """
 
         return self.selenium.find_element(By.CSS_SELECTOR, ':root')
+
+    def wait(self, timeout=10):
+        """
+        Selenium wait method
+
+        :param timeout: Sets a custom timeout length. Defaults to 10 seconds.
+        """
+        return WebDriverWait(self.selenium, timeout)
```

Comment:
The idea we talked about was to create an attribute off the base window:

```python
self.wait = WebDriverWait(self.selenium, 10)
```

This will provide a convenient default wait object.